test/NuGet.Clients.Tests/NuGet.PackageManagement.UI.Test/ViewModels/ReadmePreviewViewModelTests.cs

```diff
@@ -61,22 +63,66 @@ public async Task LoadReadmeAsync_WithoutReadmeUrl_NoReadmeReturned(string readm
         }
 
         [Fact]
-        public async Task LoadReadmeAsync_WithReadmeUrl_ReadmeReturned()
+        public async Task LoadReadmeAsync_WithLocalReadmeUrl_RenderLocalReadmeTrue_ReadmeReturned()
         {
             //Arrange
             var readmeContents = "readme contents";
             using Stream stream = new MemoryStream(Encoding.UTF8.GetBytes(readmeContents));
             var mockFileService = new Mock<INuGetPackageFileService>();
             mockFileService.Setup(x => x.GetReadmeAsync(It.IsAny<Uri>(), It.IsAny<CancellationToken>())).ReturnsAsync(stream);
             var target = new ReadmePreviewViewModel(mockFileService.Object);
+            var package = new DetailedPackageMetadata();
+            package.ReadmeFileUrl = "C://path/to/readme.md";
 
             //Act
-            await target.LoadReadmeAsync("C://path/to/readme.md", CancellationToken.None);
+            await target.LoadReadmeAsync(package, true, CancellationToken.None);
+
 
             //Assert
             Assert.False(target.ErrorLoadingReadme);
             Assert.Equal(readmeContents, target.ReadmeMarkdown);
             Assert.True(target.CanDetermineReadmeDefined);
         }
+
+        [Fact]
+        public async Task LoadReadmeAsync_WithLocalReadmeUrl_RenderLocalReadmeFalse_ReadmeReturned()
+        {
+            //Arrange
+            var readmeContents = "readme contents";
+            using Stream stream = new MemoryStream(Encoding.UTF8.GetBytes(readmeContents));
+            var mockFileService = new Mock<INuGetPackageFileService>();
+            mockFileService.Setup(x => x.GetReadmeAsync(It.IsAny<Uri>(), It.IsAny<CancellationToken>())).ReturnsAsync(stream);
+            var target = new ReadmePreviewViewModel(mockFileService.Object);
+            var package = new DetailedPackageMetadata();
+            package.ReadmeFileUrl = "C://path/to/readme.md";
+
+            //Act
+            await target.LoadReadmeAsync(package, false, CancellationToken.None);
+
+
+            //Assert
+            Assert.False(target.ErrorLoadingReadme);
+            Assert.Equal(string.Empty, target.ReadmeMarkdown);
+            Assert.False(target.CanDetermineReadmeDefined);
+        }
+
+        [Fact]
+        public async Task LoadReadmeAsync_WithLocalReadmeUrl_FileNotFound_Null()
```

Comment:
File Not Found to me should be an error - why is ErrorLoadingReadme false?

Name doesn't match assertion:
```suggestion
        public async Task LoadReadmeAsync_WithLocalReadmeUrl_FileNotFound_EmptyReadmeMarkdown()
```

Reply:
The NuGetPackageFileService returns null if the file isn't found, which I think is okay. There isn't anything for the user to do if a README is defined in the nuspec but not actually in the package. Giving them the message that there is no readme and to contact the package owner is the best course of action. That said I updated the test name